src/main/java/org/prebid/server/bidder/sharethrough/SharethroughMarkupUtil.java

```diff
@@ -0,0 +1,64 @@
+package org.prebid.server.bidder.sharethrough;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.vertx.core.json.Json;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.sharethrough.model.StrUriParameters;
+import org.prebid.server.bidder.sharethrough.model.bidResponse.ExtImpSharethroughResponse;
+
+import java.util.Base64;
+
+class SharethroughMarkupUtil {
+
+    private SharethroughMarkupUtil() {
+    }
+
+    static String getAdMarkup(ExtImpSharethroughResponse strResponse, StrUriParameters strUriParameters) {
+        final StringBuilder tmplBody = new StringBuilder();
+
+        final String strRespId = String.format("str_response_%s", strResponse.getBidId());
+        final String arId = strResponse.getAdserverRequestId();
+        final String jsonStrResponse = parseResponseToString(strResponse);
+        if (StringUtils.isBlank(jsonStrResponse)) {
+            return "";
+        }
+
+        final String encodedJsonResponse = Base64.getEncoder().encodeToString(jsonStrResponse.getBytes());
+
+        tmplBody.append("<img src=\"//b.sharethrough.com/butler?type=s2s-win&arid=").append(arId).append("\" />\n")
```

Comment:
I was thinking of maybe doing it with `String.format` method since there are only 4 variables getting inserted, but in that case it would definitely harder to read, so I liked your `StringBuilder` even more. Will see if @rpanchyk is okay with it.

Reply:
i'm ok with that.